### PR TITLE
fix(DX): next config warning

### LINKED ISSUE #340: [DX] Invalid next.config.js
When running `yarn dev`, following warning is reported:

```
warn  - Invalid next.config.js options detected: 
  - The root value has an unexpected property, webpackDevMiddleware, which is not in the list of allowed properties (amp, analyticsId, assetPrefix, basePath, cleanDistDir, compiler, compress, crossOrigin, devIndicators, distDir, env, eslint, excludeDefaultMomentLocales, experimental, exportPathMap, future, generateBuildId, generateEtags, headers, httpAgentOptions, i18n, images, onDemandEntries, optimizeFonts, output, outputFileTracing, pageExtensions, poweredByHeader, productionBrowserSourceMaps, publicRuntimeConfig, reactStrictMode, redirects, rewrites, sassOptions, serverRuntimeConfig, staticPageGenerationTimeout, swcMinify, trailingSlash, typescript, useFileSystemPublicRoutes, webpack).
  - The root value has an unexpected property, configOrigin, which is not in the list of allowed properties (amp, analyticsId, assetPrefix, basePath, cleanDistDir, compiler, compress, crossOrigin, devIndicators, distDir, env, eslint, excludeDefaultMomentLocales, experimental, exportPathMap, future, generateBuildId, generateEtags, headers, httpAgentOptions, i18n, images, onDemandEntries, optimizeFonts, output, outputFileTracing, pageExtensions, poweredByHeader, productionBrowserSourceMaps, publicRuntimeConfig, reactStrictMode, redirects, rewrites, sassOptions, serverRuntimeConfig, staticPageGenerationTimeout, swcMinify, trailingSlash, typescript, useFileSystemPublicRoutes, webpack).
  - The root value has an unexpected property, target, which is not in the list of allowed properties (amp, analyticsId, assetPrefix, basePath, cleanDistDir, compiler, compress, crossOrigin, devIndicators, distDir, env, eslint, excludeDefaultMomentLocales, experimental, exportPathMap, future, generateBuildId, generateEtags, headers, httpAgentOptions, i18n, images, onDemandEntries, optimizeFonts, output, outputFileTracing, pageExtensions, poweredByHeader, productionBrowserSourceMaps, publicRuntimeConfig, reactStrictMode, redirects, rewrites, sassOptions, serverRuntimeConfig, staticPageGenerationTimeout, swcMinify, trailingSlash, typescript, useFileSystemPublicRoutes, webpack).
  - The root value has an unexpected property, webpack5, which is not in the list of allowed properties (amp, analyticsId, assetPrefix, basePath, cleanDistDir, compiler, compress, crossOrigin, devIndicators, distDir, env, eslint, excludeDefaultMomentLocales, experimental, exportPathMap, future, generateBuildId, generateEtags, headers, httpAgentOptions, i18n, images, onDemandEntries, optimizeFonts, output, outputFileTracing, pageExtensions, poweredByHeader, productionBrowserSourceMaps, publicRuntimeConfig, reactStrictMode, redirects, rewrites, sassOptions, serverRuntimeConfig, staticPageGenerationTimeout, swcMinify, trailingSlash, typescript, useFileSystemPublicRoutes, webpack).
  - The root value has an unexpected property, include, which is not in the list of allowed properties (amp, analyticsId, assetPrefix, basePath, cleanDistDir, compiler, compress, crossOrigin, devIndicators, distDir, env, eslint, excludeDefaultMomentLocales, experimental, exportPathMap, future, generateBuildId, generateEtags, headers, httpAgentOptions, i18n, images, onDemandEntries, optimizeFonts, output, outputFileTracing, pageExtensions, poweredByHeader, productionBrowserSourceMaps, publicRuntimeConfig, reactStrictMode, redirects, rewrites, sassOptions, serverRuntimeConfig, staticPageGenerationTimeout, swcMinify, trailingSlash, typescript, useFileSystemPublicRoutes, webpack).
  - The value at .amp.canonicalBase must be 1 character or more but it was 0 characters.
  - The value at .assetPrefix must be 1 character or more but it was 0 characters.
  - The value at .experimental.outputFileTracingRoot must be 1 character or more but it was 0 characters.
  - The value at .i18n must be an object but it was null.
  - The value at .images.path must be 1 character or more but it was 0 characters.

See more info here: https://nextjs.org/docs/messages/invalid-next-config
```

### PR DESCRIPTION
**Pull request by AleksandarDev**
Closes #340

### CODE REVIEW

File: hooks/useDarkMode.js [hunk 1]
```diff
@@ -13,16 +10,16 @@ export default function useDarkMode() {
...
 
   useEffect(() => {
-    ThemeService.setIsDarkMode(isDarkMode);
+    persistIsDarkMode(isDarkMode);
   }, [isDarkMode]);
 
   const toggleThemeChange = useCallback(() => {
     setIsDarkMode((prev) => !prev);
-  });
+  }, []);
```
Comment (radovix) [inline hooks/useDarkMode.js:22]: Do we even need `useCallback` here?

File: services/LocalStorageService.js [hunk 2]
```diff
@@ -1,52 +1,47 @@
...
-   * @returns value associated with item name or null if local storage is not defined
-   * @memberof LocalStorageService
-   */
-  static get(itemName) {
-    if (typeof localStorage !== 'undefined') {
-      return JSON.parse(localStorage.getItem(itemName));
+export function getLocalStorageItem(itemName) {
+  if (typeof localStorage !== 'undefined') {
+    const value = localStorage.getItem(itemName);
+    if (value == null || value === 'undefined' || value === 'null') {
```
Comment (radovix) [inline services/LocalStorageService.js:11]: And what if somebody really wanted to save the value "null"? 😅

File: services/NotificationsService.js [hunk 3]
```diff
@@ -8,141 +7,97 @@ Noty.overrideDefaults({
...
 
-export default NotificationService;
+/**
+ * Shows the default error notification.
+ *
+ * @static
+ * @returns notification
+ */
+export function showDefaultErrorNotification() {
+  return showErrorNotification('Trenutno nije moguće dohvatiti podatke. Osvježite stranicu da biste pokušali ponovno.');
```
Comment (radovix) [inline services/NotificationsService.js:102]: Maybe we should translate this.

File: helpers/ComponentsHelper.js [hunk 4]
```diff
@@ -40,7 +40,8 @@ export const withAppModels = (Component, AppModelProviders) => function WithAppM
   }, null);
 };
 
-export default {
+const exports = {
   withViewModel,
   withAppModels
 };
+export default exports;
```
Comment (radovix) [inline helpers/ComponentsHelper.js:47]: Why did you rewrite it this way?

File: hooks/useDarkMode.js (hunk 1 above)
Comment (AleksandarDev) [reply hooks/useDarkMode.js:22]: Didn't touch that part, only fixed ESLint issue of missing hook deps.

File: services/LocalStorageService.js (hunk 2 above)
Comment (AleksandarDev) [reply services/LocalStorageService.js:11]: This is a get function - it would return null as expected

File: helpers/ComponentsHelper.js (hunk 4 above)
Comment (AleksandarDev) [reply helpers/ComponentsHelper.js:47]: ESLint was failing here

File: services/NotificationsService.js (hunk 3 above)
Comment (AleksandarDev) [reply services/NotificationsService.js:102]: Yeah, you can create issue to translate whole application (check all and translate to en).

File: helpers/ComponentsHelper.js (hunk 4 above)
Comment (AleksandarDev) [reply helpers/ComponentsHelper.js:47]: https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-anonymous-default-export.md

**Pull request merged**